src/widgets/RRule.tsx

```diff
@@ -0,0 +1,284 @@
+  onChange: (rrule: string) => void;
+  rrule: string;
+}
+enum Ends {
+  Never,
+  Date,
+  After,
+}
+enum Months {
+  Jan = 1,
+  Feb,
+  Mar,
+  Apr,
+  May,
+  Jun,
+  Jul,
+  Aug,
+  Sep,
+  Oct,
+  Nov,
+  Dec,
+}
+enum MonthReapet {
+  bysetpos,
+  bymonthday,
+}
+enum Position {
+  First = 1,
+  Second = 2,
+  Third = 3,
+  Fourth = 4,
+  Last = -1,
+}
+const menueItemsPosition = Object.keys(Position).filter((key) => Number(key)).map((key) => {
+  return (
+    <MenuItem key={key} value={key}>{Position[key]}</MenuItem>
+  );
+});
+const menueItemsMonths = Object.keys(Months).filter((key) => Number(key)).map((key) => {
+  return (
+    <MenuItem key={key} value={key}>{Months[key]}</MenuItem>
+  );
+});
+const menueItemsEnds = [Ends.Never, Ends.Date, Ends.After].map((key) => {
+  return (
+    <MenuItem key={key} value={key}>{Ends[key]}</MenuItem>
+  );
+});
+const bysetposWeekDay = ALL_WEEKDAYS.map((value, index) => {
+  return (
+    <MenuItem key={index} value={index}>{value}</MenuItem>
+  );
+});
+const menuItemsFrequency = Object.keys(Frequency).filter((key) => Number(key) < 4).map((value) => {
+  return (
+    <MenuItem key={value} value={value}>{Frequency[value].toLowerCase()}</MenuItem>
+  );
+});
+const menuItemDaysNumbers = Array.from(Array(31).keys()).map((value: number) => {
+  return (
+    <MenuItem key={value} value={value + 1}>{value + 1}</MenuItem>
+  );
+});
+
+export default function RRuleEteSync(props: PropsType) {
+  const options = RRule.fromString(props.rrule).origOptions;
+
+  function updateRule(newOptions: Partial<Options>): void {
+    const updatedOptions = { ...options, ...newOptions };
+    props.onChange((new RRule(updatedOptions)).toString());
+  }
+  function getEnds(): Ends {
+    if (options.until && !options.count) {
+      return Ends.Date;
+    } else if (!options.until && options.count) {
+      return Ends.After;
+    } else {
+      return Ends.Never;
+    }
+  }
+  function handleCheckboxWeekday(event: React.FormEvent<{ value: unknown }>): void {
+    const checkbox = event.target as HTMLInputElement;
+    const weekday = new Weekday(Number(checkbox.value));
+    let byweekdayArray = options.byweekday as ByWeekday[];
+    let byweekday;
+    if (!checkbox.checked && byweekdayArray) {
+      byweekday = byweekdayArray.filter((day) => (day as Weekday).weekday !== weekday.weekday);
+    } else if (byweekdayArray) {
+      byweekdayArray = byweekdayArray.filter((day) => (day as Weekday).weekday !== weekday.weekday);
+      byweekday = [...byweekdayArray, weekday];
+    } else {
+      byweekday = weekday;
+    }
+    updateRule({ byweekday: byweekday });
+  }
+  function isWeekdayChecked(day: number): boolean {
+    const weekdayArray = options.byweekday as ByWeekday[];
+    if (weekdayArray) {
+      return !!(weekdayArray.find((value) => (value as Weekday).weekday === day));
+    } else {
+      return false;
+    }
+  }
+  const checkboxWeekDays = ALL_WEEKDAYS.map((value, index) => {
```

Comment:
Super nit-picky: annoying visually that here it's "WeekDay" and everywhere else is "Weekday". It should be consistent (probably as the latter...)